### ISSUE TITLE
Crash SDWebImageDownloaderOperation.m line 247

### ISSUE THREAD

**Issue by guidev** [SDWebImage/SDWebImage#3433]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios
 Platform Version        | 16.0
 SDWebImage Version      | 5.14.0
 Integration Method      | spm
 Xcode Version           | Xcode 
 Repro rate              | sometimes (2%)
 Repro with our demo prj | not tested
 Demo project link       | n/a

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

I'm seeing lots of crashes on Firebase since updating to `v 5.14.0`

<img width="1051" alt="Screenshot 2022-11-09 at 15 44 17" src="https://user-images.githubusercontent.com/1105813/200860811-c5cfd692-ad30-4429-af0d-a0693a0b20aa.png">

**Comment by dreampiggy** [comment]
Maybe related to the refactory to #3423, thread safe issue ?

**Comment by dreampiggy** [comment]
Someone told me that NSOperation.underlyingQueue is nil, will cause runtime crash.

![img_v2_5555cf33-5064-4067-ad84-8437a70323ag](https://user-images.githubusercontent.com/6919743/201053896-6dae4b0a-b584-46d1-b91b-f1a85c27f825.jpg)

**Comment by guidev** [comment]
<img width="1195" alt="Screenshot 2022-11-10 at 17 28 47" src="https://user-images.githubusercontent.com/1105813/201151988-5c019a0a-2fd9-4188-a482-955974f4f53b.png">


I haven't checked if #3434 fixes this issue, but my app is available only for > iOS 14, so I don't think that's the issue

**Comment by guidev** [comment]
<img width="1673" alt="Screenshot 2022-11-10 at 17 40 04" src="https://user-images.githubusercontent.com/1105813/201154513-aa978796-b18b-4a33-878d-52a8dd55dc7c.png">

This is from Xcode crash reports

**Comment by dreampiggy** [comment]
These new screenshot seems another issue

**Comment by dreampiggy** [comment]
@guidev These two crash should be fixed in [5.14.1](https://github.com/SDWebImage/SDWebImage/releases/tag/5.14.1)

**Comment by guidev** [comment]
@dreampiggy awesome, thanks!

**Comment by JadianZheng** [comment]
It was luck to find this. Meet this issue when build the iOS project after upgrade macOS to 13.0.1 .

**Comment by devssun** [comment]
Hello, I have a same issue on iOS 12.
- 95% of crash events for this issue occurred within the first 5 seconds of starting a user session.
- Crash events corresponding to this issue occurred on average 3 times per user.

I run on iOS 12.4 simulator but it works, so I can't attach Xcode crash reports.
Is this issue resolved at 5.14.1 version? please check.

### Issue Info

Info | Value
-- | --
Platform Name | ios, ipad
Platform Version | 12.5.6, 12.4.1, 12.5.5
SDWebImage Version | 5.14.0
Integration Method | spm
Xcode Version | Xcode 14.1
Repro rate | sometimes (2%)
Repro with our demo prj | not tested
Demo project link | n/a

<br class="Apple-interchange-newline">

### error log
```
Crashed: NSOperationQueue 0x282e8ebc0 (QOS: UNSPECIFIED)
0  libdispatch.dylib              0x62f8 dispatch_barrier_async$VARIANT$mp + 160
1  AppName                       0x32fabc -[SDWebImageDownloaderOperation URLSession:task:didCompleteWithError:] + 614 (SDWebImageDownloaderOperation.m:614)
2  AppName                       0x32bbd8 -[SDWebImageDownloader URLSession:task:didCompleteWithError:] + 505 (SDWebImageDownloader.m:505)
3  FirebasePerformance            0x17298 __InstrumentURLSessionTaskDidCompleteWithError_block_invoke + 208
4  CFNetwork                      0x1df420 __51-[NSURLSession delegate_task:didCompleteWithError:]_block_invoke.182 + 252
5  Foundation                     0x112cf4 __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 16
6  Foundation                     0x1ed90 -[NSBlockOperation main] + 72
7  Foundation                     0x1e278 -[__NSOperationInternal _start:] + 740
8  Foundation                     0x114a78 __NSOQSchedule_f + 272
9  libdispatch.dylib              0x5fa38 _dispatch_call_block_and_release + 24
10 libdispatch.dylib              0x607d4 _dispatch_client_callout + 16
11 libdispatch.dylib              0x501c _dispatch_continuation_pop$VARIANT$mp + 412
12 libdispatch.dylib              0x46e0 _dispatch_async_redirect_invoke + 600
13 libdispatch.dylib              0x11030 _dispatch_root_queue_drain + 372
14 libdispatch.dylib              0x118d4 _dispatch_worker_thread2 + 128
15 libsystem_pthread.dylib        0xc1b4 _pthread_wqthread + 464
16 libsystem_pthread.dylib        0xecd4 start_wqthread + 4
```
